ambry-network/src/main/java/com.github.ambry.network/ConnectionTracker.java

```diff
@@ -25,28 +31,27 @@
  */
 
 class ConnectionTracker {
-  private final HashMap<String, HostPortPoolManager> hostPortToPoolManager;
-  private final HashMap<String, HostPortPoolManager> connectionIdToPoolManager;
+  private static final Logger LOGGER = LoggerFactory.getLogger(ConnectionTracker.class);
+  private final HashMap<Pair<String, Port>, HostPortPoolManager> hostPortToPoolManager = new HashMap<>();
+  private final HashMap<String, HostPortPoolManager> connectionIdToPoolManager = new HashMap<>();
+  private final HashSet<HostPortPoolManager> poolManagersBelowWatermark = new HashSet<>();
```

Comment:
`watermark` is sort of odd terminology to me, that come from something?  I would think something like `poolManagersBelowMinimumActiveConnections` would make more sense

Reply:
I think the netty code uses watermark in a few places, but mostly when dealing with buffers, not connections (https://netty.io/4.1/api/io/netty/channel/WriteBufferWaterMark.html). I'm not too tied to the terminology though.